astropy/coordinates/angles.py

```diff
@@ -325,7 +331,8 @@ def to_string(
                 if format == "latex" or format == "latex_inline":
                     unit_string = unit_string[1:-1]
                 format_func = func
-                func = lambda x: format_func(x) + unit_string
+                # Add a space if no format(i.e unicode or latex) is given
```

Comment:
I wonder if the whole `else` clause could simply leave it to `Quantity.to_string()` via
```
return super().to_string(unit=unit, precision=precision, format=format)
```